packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -617,27 +617,7 @@ function getSchemaForModel(
     modelSchema.properties[name] = newPropSchema;
   }
 
-  // Special case: if a model type extends a single *templated* base type and
-  // has no properties of its own, absorb the definition of the base model
-  // into this schema definition.  The assumption here is that any model type
-  // defined like this is just meant to rename the underlying instance of a
-  // templated type.
-  if (
-    model.baseModel &&
-    model.baseModel.templateMapper &&
-    model.baseModel.templateMapper.args &&
-    model.baseModel.templateMapper.args.length > 0 &&
-    modelSchema.properties &&
-    Object.keys(modelSchema.properties).length === 0
-  ) {
-    // Take the base model schema but carry across the documentation property
-    // that we set before
-    const baseSchema = getSchemaForType(dpgContext, model.baseModel, usage);
-    modelSchema = {
-      ...baseSchema,
-      description: modelSchema.description
-    };
-  } else if (model.baseModel) {
+  if (model.baseModel) {
```

Comment:
Could you help me understand why we have this change? We don't need to handle template model by ourselves?

Reply:
In the case of `model A extends Record<B> {}`, we will run into this condition, which will return a model without name, and we will skip the generation of the model A. As I really don't know what this is trying to do, and I found there's no such logic in OpenAPI3 emitter, so just removed it and found nowhere is reporting errors about it.